### PR TITLE
Profile screen

### LINKED ISSUE #3: Profile Screen
- Create Profile Screen for users to view information about yourselfs;
- Implement functions to allow users edit their informations;

### PR DESCRIPTION
**Pull request by newtonneto**
Allow users to update their own profiles

### CODE REVIEW

File: src/utils/getImageType.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const getImageType = (item: String): String => {
+  const splitedItem = item.split('.');
+
+  return splitedItem[splitedItem.length - 1];
```
Comment (breno-alves) [inline src/utils/getImageType.ts]:
```suggestion
  return splitedItem.pop();
```
this solution works even if `item.pop()` is cleaner than using array indexs

File: src/utils/getPictureFromStorage.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+const getPictureFromStorage = async (): Promise<ImagePickerResponse> => {
+  const options: ImageLibraryOptions = {
+    quality: 1,
+    mediaType: 'photo',
+  };
+
+  const result = await launchImageLibrary(options);
+
+  return result;
```
Comment (breno-alves) [inline src/utils/getPictureFromStorage.ts]:
```suggestion
  return launchImageLibrary(options);
```

redundant `await`

File: src/utils/handleGalleryPermission.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+};
+
+const handleGalleryPermissions = async (): Promise<PermissionStatus> => {
+  let result: PermissionStatus;
+
+  if (Platform.OS === 'android') {
+    result = await check(PERMISSIONS.ANDROID.READ_EXTERNAL_STORAGE);
+  } else {
+    result = await check(PERMISSIONS.IOS.PHOTO_LIBRARY);
+  }
```
Comment (breno-alves) [inline src/utils/handleGalleryPermission.ts]:
```suggestion
  result = Platform.OS === 'android' ? 
    await check(PERMISSIONS.ANDROID.READ_EXTERNAL_STORAGE) :
    await check(PERMISSIONS.IOS.PHOTO_LIBRARY);
```

File: src/utils/handleGalleryPermission.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      );
+      break;
+    case RESULTS.DENIED:
+      (await request(
+        Platform.OS === 'android'
+          ? PERMISSIONS.ANDROID.READ_EXTERNAL_STORAGE
+          : PERMISSIONS.IOS.PHOTO_LIBRARY,
+      )) === 'blocked' &&
+        Platform.OS === 'android' &&
+        blockedPermission();
```
Comment (breno-alves) [inline src/utils/handleGalleryPermission.ts]: this part of code is really complex can we change that somehow?

File: src/utils/inputMasks.ts [hunk 5]
```diff
@@ -19,33 +19,13 @@ export const handleDateMask = (date: string): string => {
...
-    maskedPhone = `${splitedPhone[0]} ${splitedPhone[1]}`;
-  }
-
-  if (phone.length === 12 && phone[11] !== '-') {
-    const splitedPhone: string[] = [phone.slice(0, 11), phone.slice(11)];
-    maskedPhone = `${splitedPhone[0]}-${splitedPhone[1]}`;
-  }
+  phone = phone.replace(/\D/g, '');
+  phone = phone.replace(/^(\d\d)(\d)/g, '($1) $2');
+  phone = phone.replace(/(\d{5})(\d)/, '$1-$2');
```
Comment (breno-alves) [inline src/utils/inputMasks.ts]: I would add comments or variable for those regex to explain what they do exactly

**Review by breno-alves — CHANGES_REQUESTED** [review]

File: src/utils/getPictureFromStorage.ts (hunk 2 above)
Comment (newtonneto) [reply src/utils/getPictureFromStorage.ts:15]: Done

File: src/utils/getImageType.ts (hunk 1 above)
Comment (newtonneto) [reply src/utils/getImageType.ts]: Done

File: src/utils/handleGalleryPermission.ts (hunk 3 above)
Comment (newtonneto) [reply src/utils/handleGalleryPermission.ts]: Done

File: src/utils/handleGalleryPermission.ts (hunk 4 above)
Comment (newtonneto) [reply src/utils/handleGalleryPermission.ts]: I improved code legibility

File: src/utils/inputMasks.ts (hunk 5 above)
Comment (newtonneto) [reply src/utils/inputMasks.ts]: Done

File: src/utils/getPictureFromStorage.ts (hunk 2 above)
Comment (newtonneto) [reply src/utils/getPictureFromStorage.ts]: Done

**Review by paullarachel — APPROVED** [review]

**Review by breno-alves — APPROVED** [review]

**Pull request merged**